src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1015,3 +1057,917 @@ def upgrade_agents(cmd, client, resource_group_name, cluster_name, kube_config=N
+        raise CLIError("Failed to merge kubeconfig." + str(e))
+
+
+def make_api_call_with_retries(uri, data, tls_verify, fault_type, summary, cli_error):
+    for i in range(consts.API_CALL_RETRIES):
+        try:
+            response = requests.post(uri, json=data, verify=tls_verify)
+            return response
+        except Exception as e:
+            if i != consts.API_CALL_RETRIES - 1:
+                pass
+            else:
+                telemetry.set_exception(exception=e, fault_type=fault_type, summary=summary)
+                raise CLIError(cli_error + str(e))
+
+
+def insert_token_in_kubeconfig(data, token):
+    b64kubeconfig = data['kubeconfigs'][0]['value']
+    decoded_kubeconfig_str = b64decode(b64kubeconfig).decode("utf-8")
+    dict_yaml = yaml.safe_load(decoded_kubeconfig_str)
+    dict_yaml['users'][0]['user']['token'] = token
+    kubeconfig = yaml.dump(dict_yaml).encode("utf-8")
+    b64kubeconfig = b64encode(kubeconfig).decode("utf-8")
+    return b64kubeconfig
+
+
+def check_process(processName):
+    '''
+    Check if there is any running process that contains the given name processName.
+    '''
+    for proc in process_iter():
+        try:
+            if proc.name().startswith(processName):
+                return True
+        except (NoSuchProcess, AccessDenied, ZombieProcess):
+            pass
+    return False
+
+
+def ctrlc_handler(sig, frame):
+    interrupt_main()
+
+
+def get_custom_locations_oid(cmd):
+    try:
+        sp_graph_client = get_graph_client_service_principals(cmd.cli_ctx)
+        sub_filters = []
+        sub_filters.append("displayName eq '{}'".format("Custom Locations RP"))
+        result = list(sp_graph_client.list(filter=(' and '.join(sub_filters))))
+        if len(result) != 0:
+            return result[0].object_id
+        else:
+            logger.warning("Unable to fetch oid of 'custom-locations' app. Proceeding without enabling the feature.")
+            telemetry.set_exception(exception='Unable to fetch oid of custom locations app.', fault_type=consts.Custom_Locations_OID_Fetch_Fault_Type,
+                                    summary='Unable to fetch oid for custom locations app.')
+            return ""
+    except Exception as e:
+        logger.warning("Unable to fetch oid of 'custom-locations' app. Proceeding without enabling the feature. " + str(e))
+        telemetry.set_exception(exception=e, fault_type=consts.Custom_Locations_OID_Fetch_Fault_Type,
+                                summary='Unable to fetch oid for custom locations app.')
+        return ""
+
+
+def check_cl_registration_and_get_oid(cmd):
+    enable_custom_locations = True
+    custom_locations_oid = ""
+    try:
+        rp_client = _resource_providers_client(cmd.cli_ctx)
+        cl_registration_state = rp_client.get(consts.Custom_Locations_Provider_Namespace).registration_state
+        if cl_registration_state != "Registered":
+            enable_custom_locations = False
```

Comment:
Lets add telemetry exception here with specific fault type since we will raise error above (when  user is enabling only CL & this is set to false)